src/sync/channel.zig

```diff
@@ -128,19 +143,27 @@ pub fn Channel(T: type) type {
 
         /// to deinit channels created with `create`
         pub fn destroy(channel: *Self) void {
+            const allocator = channel.allocator; // read allocator from channel before freeing it
```

Comment:
Why is this necessary? `deinit` does not impact the allocator field.

Reply:
https://github.com/Syndica/sig/pull/486/commits/767607c3c46beb88cf4533d9bbabba71b9b3e634